### PR TITLE
fix: enable sql charts' csv exports and schedules

### PR DESCRIPTION
**Pull request by joaoviana**
Closes: [#11655](https://github.com/lightdash/lightdash/issues/11655)

### Description:

- Waits for more endpoints (sql-chart-related) before taking screenshot
- Gets csvs for sql charts in dashboards:

**on export**
Screenshot 2024-09-27 at 14 20 14
Screenshot 2024-09-27 at 14 20 19

**on scheduled delivery**
Screenshot 2024-09-27 at 14 22 34
Screenshot 2024-09-27 at 14 23 01

See PR comments ⬇️ 

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/backend/src/services/UnfurlService/UnfurlService.ts [hunk 1]
```diff
@@ -217,6 +217,7 @@ export class UnfurlService extends BaseService {
             chartType,
             resourceUuid,
             chartTileUuids: rest.chartTileUuids,
+            sqlChartTileUuids: rest.sqlChartTileUuids,
```
Comment (joaoviana) [inline packages/backend/src/services/UnfurlService/UnfurlService.ts:220]: was missing this value being passed to the main screenshot. Was only waiting properly for sql charts when there was a retryable error, like protocol error, or timeout error

File: packages/backend/src/services/UnfurlService/UnfurlService.ts [hunk 2]
```diff
@@ -505,12 +506,25 @@ export class UnfurlService extends BaseService {
...
                                         page?.waitForResponse(
-                                            /\/sqlRunner\/results/,
+                                            new RegExp(
+                                                `/sqlRunner/saved/${id}`,
+                                            ),
+                                            {
+                                                timeout: 60000,
+                                            },
+                                        ), // NOTE: No await here
+                                );
```
Comment (joaoviana) [inline packages/backend/src/services/UnfurlService/UnfurlService.ts:522]: waits for 2 response regexes
- chart results (like name, config, etc)
- chart query results (against the warehouse)

Note: we can't combine these 2 bc playwright won't await for them properly

**Comment by rephus** [comment]
Is this the right issue ? 

> Closes: https://github.com/lightdash/lightdash/issues/11655
The issue is about CSV downloads, not about screenshots.

**Comment by joaoviana** [comment]
> Is this the right issue ?
> 
> > Closes: #11655
> 
> The issue is about CSV downloads, not about screenshots.

🤦 
meant to do it on this one but was too focused on screenshots 
Will add more to this - converting to draft!

File: packages/backend/src/services/CsvService/CsvService.ts [hunk 3]
```diff
@@ -956,15 +1118,13 @@ export class CsvService extends BaseService {
                 void archive.finalize(); // This finalize doesn't wait for the files to be written
             });
 
-        const csvFiles = await this.getCsvsForDashboard(
+        const csvFiles = await this.getCsvsForDashboard({
```
Comment (joaoviana) [inline packages/backend/src/services/CsvService/CsvService.ts:1121]: was getting too hard to maintain this function with many args, so I changed it to a single object

File: packages/frontend/src/components/DataViz/hooks/useChartViz.ts [hunk 4]
```diff
@@ -115,23 +117,35 @@ export const useChartViz = <T extends ResultsRunner>({
...
-            return chartDataModel.getEchartsSpec(
-                transformedData,
-                config.display,
-            );
+            return {
+                ...chartDataModel.getEchartsSpec(
+                    transformedData,
+                    config.display,
+                ),
+                animation: !inDashboard,
```
Comment (joaoviana) [inline packages/frontend/src/components/DataViz/hooks/useChartViz.ts:125]: turn off animation if on dashboard - we do the same for explore charts

File: packages/backend/src/services/CsvService/CsvService.ts [hunk 5]
```diff
@@ -481,53 +547,130 @@ export class CsvService extends BaseService {
...
+                userId: user.userUuid,
+                properties: {
+                    ...analyticProperties,
+                    numRows: resultsFileUrl.rows.length,
+                    numColumns: Object.keys(resultsFileUrl.fields).length,
+                },
+            });
+        }
+
+        return this.downloadCsvFile({
```
Comment (joaoviana) [inline packages/backend/src/services/CsvService/CsvService.ts:648]: reusable util for both types of charts

**Review by rephus — COMMENTED** [review]
I tested in render

It works great for CSVs

But for images, I never got the email (I tried a few times) 
I saw this error in render logs , not sure if related.
image

For this dashboard with only 1 sql runner chart
https://lightdash-pr-11701.onrender.com/projects/3675b69e-8324-4110-bdca-059031aa8da3/dashboards/ecbd687a-d256-4780-a1fa-3e0a1241f2c2/view

**Comment by joaoviana** [comment]
@rephus , soon Render will be ready - and this only covers the csv work. 

I've opened a separate bug ticket for scheduled delivery images: https://github.com/lightdash/lightdash/issues/11724

**Review by rephus — APPROVED** [review]
CSV on deliveries worked, 

On image, I got this (again), I guess that is what you were trying to fix 

image

**Pull request merged**